src/components/UTIL_FormField.component

```diff
@@ -0,0 +1,113 @@
+                                        'multipicklist', 'slds-select multipicklistSLDSFixup',
+                                        'reference', IF(field=='RecordTypeId', 'slds-select', 'lookupInput ' + field),
+                                        'slds-input')}">
+                <apex:actionSupport event="onchange" action="{!actSupAction}" reRender="{!actSupReRender}" rendered="{!actSup}" status="{!actSupStatus}" immediate="{!actSupImmediate}"/>
+            </apex:inputField>
+        </div>
+        <!-- Javascript will add the onchange event when needed and swap out the icon for lookup fields -->
+        <script type="text/javascript">
+            var oc = '{!onchange}';
+            var ftype = '{!ftype}';
+            var field = '{!field}';
+            var input = document.getElementById("{!$Component.inputX}");
+
+            // Add onchange event only when needed
+            if (oc) input.setAttribute("onchange", oc);
+
+            // Swap out the search icon
+            if (ftype == 'reference' && field != 'RecordTypeId') {
+                input.style.visibility = "";
+                var lkSpan = input.parentElement;
+                var lkLink = lkSpan.querySelector("a");
+                lkLink.style.visibility = "";
+                lkLink.className = "";
+                lkLink.innerHTML = "<svg aria-hidden=\"true\" class=\"slds-icon slds-input__icon slds-icon--x-small slds-icon-text-default\" viewBox=\"0 0 24 24\"><path  d=\"M22.9 20.9l-6.2-6.1c1.3-1.8 1.9-4 1.6-6.4-.6-3.9-3.8-7.1-7.8-7.4C5 .4.4 5 1 10.5c.3 4 3.5 7.3 7.4 7.8 2.4.3 4.6-.3 6.4-1.5l6.1 6.1c.3.3.7.3 1 0l.9-1c.3-.3.3-.7.1-1zM3.7 9.6c0-3.2 2.7-5.9 5.9-5.9s6 2.7 6 5.9-2.7 6-6 6-5.9-2.6-5.9-6z\"></path></svg>";
+            }
+        </script>
+    </apex:outputPanel>
+
+    <!-- Updateable boolean fields need to have input within the label span. -->
+    <apex:outputPanel layout="block" styleClass="{!'slds-form-element ' + styleClass}" id="divCheckbox"
+        rendered="{!($ObjectType[sObjType].fields[field].updateable
+                    || $ObjectType[sObjType].fields[field].createable && sObj['Id']=='') && $ObjectType[sObjType].fields[field].type == 'boolean'}">
+            <div class="slds-form-element__control">
+                <apex:outputLabel styleClass="slds-checkbox" for="inputCB">
+                    <apex:inputField id="inputCB" value="{!sObj[field]}" label="">
+                        <apex:actionSupport event="onchange" action="{!actSupAction}" reRender="{!actSupReRender}" rendered="{!actSup}" status="{!actSupStatus}" immediate="{!actSupImmediate}"/>
+                    </apex:inputField>
+                        <span class="slds-checkbox--faux"></span>
+                        <span class="slds-form-element__label">{!label}</span>
+                </apex:outputLabel>
+            </div>
+            <script type="text/javascript">
+                var oc = '{!onchange}';
```

Comment:
Because this script is not wrapped in an iife, the variables defined here will be global. Same as the other. Global meaning, attached to `window`.